src/DETHRACE/common/racesumm.c

```diff
@@ -86,46 +226,207 @@ void DrawSummary(int pCurrent_choice, int pCurrent_mode) {
+                gTemp_lost -= credit_delta;
+                gTemp_credits -= credit_delta;
+                RampUpRate(&gCredits_per_ms, the_time - last_change_time);
+                if (gTemp_lost <= 0) {
+                    gTemp_credits -= gTemp_lost;
+                    S3StopOutletSound(gIndexed_outlets[0]);
+                    gSumm_sound = 0;
+                    gTemp_lost = 0;
+                    last_change_time = the_time;
+                }
+            }
+        } else if (gTemp_rank_increase != 0.f) {
+            ResetInterfaceTimeout();
+            if (the_time - last_change_time > 1000) {
+                rank_delta = (the_time - last_time) * gRank_per_ms;
+                old_temp_increase = gTemp_rank_increase;
+                gTemp_rank_increase -= rank_delta;
+                if ((int)(old_temp_increase + 0.5f) != (int)(gTemp_rank_increase + 0.5f)) {
+                    if (gTemp_rank > 1) {
+                        gTemp_rank -= 1;
+                    }
+                    gSumm_sound = DRS3StartSound(gIndexed_outlets[0], 3202);
+                }
+                if (gTemp_rank_increase <= 0.f) {
+                    S3StopOutletSound(gIndexed_outlets[0]);
+                    gSumm_sound = 0;
+                    gTemp_rank_increase = 0.f;
+                    last_change_time = the_time;
+                }
+            }
+        } else {
+            S3StopOutletSound(gIndexed_outlets[0]);
+            gSumm_sound = 0;
+        }
+    } else {
+        last_change_time = the_time;
+    }
+    DrawSummaryItems();
+    last_time = the_time;
 }
 
 // IDA: void __cdecl StartSummary()
 void StartSummary() {
     LOG_TRACE("()");
-    NOT_IMPLEMENTED();
+
+    DrawSummaryItems();
+    gSummary_start = PDGetTotalTime();
 }
 
 // IDA: void __cdecl SetUpTemps()
 void SetUpTemps() {
     LOG_TRACE("()");
-    NOT_IMPLEMENTED();
+
+    gTemp_earned = gProgram_state.credits_earned;
+    gTemp_lost = gProgram_state.credits_lost;
+    gTemp_rank_increase = gRank_increase;
+    gTemp_rank = gProgram_state.rank;
+    gTemp_credits = gProgram_state.credits;
 }
 
 // IDA: int __usercall Summ1GoAhead@<EAX>(int *pCurrent_choice@<EAX>, int *pCurrent_mode@<EDX>)
 int Summ1GoAhead(int* pCurrent_choice, int* pCurrent_mode) {
     LOG_TRACE("(%p, %p)", pCurrent_choice, pCurrent_mode);
-    NOT_IMPLEMENTED();
+
+    S3StopOutletSound(gIndexed_outlets[0]);
+    MungeRankEtc(&gProgram_state);
+    SetUpTemps();
+    DrawSummaryItems();
+    return 1;
 }
 
 // IDA: int __usercall SummCheckGameOver@<EAX>(int *pCurrent_choice@<EAX>, int *pCurrent_mode@<EDX>)
 int SummCheckGameOver(int* pCurrent_choice, int* pCurrent_mode) {
     int i;
     tS3_sound_tag sound_tag;
     LOG_TRACE("(%p, %p)", pCurrent_choice, pCurrent_mode);
-    NOT_IMPLEMENTED();
+
+    if (gTemp_credits > 0) {
+        return 0;
+    }
+    S3StopOutletSound(gIndexed_outlets[0]);
+    RemoveTransientBitmaps(1);
+    for (i = 0; i < 7; i++) {
```

Comment:
`i` isn't used anywhere in the loop?

Reply:
Nope, not used.
I think this loop is used to smoothly stop the music.